cli/src/pager.rs

```diff
@@ -0,0 +1,135 @@
+use clap::{App, Arg, ArgMatches};
+use std::str::FromStr;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct Pager {
+    size: PageSize,
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum PageSize {
+    All,
+    Size(usize),
+}
+
+impl Pager {
+    pub fn add<'a, 'b>(app: App<'a, 'b>) -> App<'a, 'b> {
+        app.arg(
+            Arg::with_name("all")
+                .long("all")
+                .conflicts_with("page-size")
+                .help("Query for all records"),
+        ).arg(
+            Arg::with_name("page-size")
+                .long("page-by")
+                .conflicts_with("all")
+                .takes_value(true)
+                .validator(Self::validator)
+                .help("Query and display in a specific page size. Defaults to 10"),
+        )
+    }
+
+    fn validator(v: String) -> Result<(), String> {
+        usize::from_str(&v)
+            .map(|_| ())
+            .map_err(|_| String::from("Page size must be a positive integer"))
+    }
+
+    pub fn from_arg<'a>(arg: &'a ArgMatches) -> Pager {
+        if let Some(size) = arg.value_of("page-size") {
+            Pager {
+                size: PageSize::Size(usize::from_str(&size).unwrap_or(10)),
+            }
+        } else if arg.is_present("all") {
+            Pager {
+                size: PageSize::All,
+            }
+        } else {
+            Pager {
+                size: PageSize::Size(10),
+            }
+        }
+    }
+
+    pub fn paginate<I, T, F>(&self, iter: I, mut f: F)
+    where
+        I: Iterator<Item = T>,
+        F: FnMut(T),
+    {
+        match self.size {
+            PageSize::All => iter.for_each(f),
+            PageSize::Size(page_size) => for (i, item) in iter.enumerate() {
+                f(item);
+
+                if (i + 1) % page_size == 0 && !Self::next_page() {
```

Comment:
It appears that when users call `All` they will not have the option to interactively paginate.  Namely `next_page()` is never referenced in the all arm.  Is this intentional or am I missing something?

Reply:
That is correct. `all` means `all`... you get the firehose.

Reply:
well, you get 200 and don't have the option to request more.  Is that correct?

Reply:
No, we request 200 at a time from horizon. You are displayed a fire hose.

Reply:
ah right, the iterator on endpoint does this for us.  Could be annoying if I called All on something like effect and just had an endless firehose, but I could also see this as being pretty useful in other contexts.  Cool.  Thanks for going through this with me.